druntime/src/core/sync/condition.d

```diff
@@ -117,7 +118,15 @@ class Condition
         }
         else version (Posix)
         {
-            m_assocMutex = m;
+            static if (is(Q == shared))
+            {
+                import core.atomic : atomicLoad;
+                m_assocMutex = atomicLoad(m);
+            }
+            else
+            {
+                m_assocMutex = m;
```

Comment:
Is this branch an optimization over always doing `m_assocMutex = atomicLoad(m);`, or does that give an error?